Zend/zend_alloc.c

```diff
@@ -1730,10 +1730,10 @@ static void *zend_mm_alloc_huge(zend_mm_heap *heap, size_t size ZEND_FILE_LINE_D
 	void *ptr;
 
 #if ZEND_MM_LIMIT
-	if (UNEXPECTED(new_size > heap->limit - heap->real_size)) {
-		if (zend_mm_gc(heap) && new_size <= heap->limit - heap->real_size) {
+	if (UNEXPECTED(new_size == 0 || new_size > heap->limit - heap->real_size)) {
```

Comment:
Add a comment for the zero check?

Reply:
It's better to check new_size for zero once and jump to the proper path.
The following two checks are not going to be necessary.

Reply:
Thanks, makes sense!